Content.Server/Body/Systems/InternalsSystem.cs

```diff
@@ -178,28 +178,22 @@ private void OnInhaleLocation(Entity<InternalsComponent> ent, ref InhaleLocation
             _alerts.ShowAlert(ent, ent.Comp.InternalsAlert, GetSeverity(ent));
         }
     }
-    public void DisconnectBreathTool(Entity<InternalsComponent> ent)
+    public void DisconnectBreathTool(Entity<InternalsComponent> ent, EntityUid toolEntity)
     {
-        var old = ent.Comp.BreathToolEntity;
-        ent.Comp.BreathToolEntity = null;
+        ent.Comp.BreathTools.Remove(toolEntity);
 
-        if (TryComp(old, out BreathToolComponent? breathTool))
-        {
-            _atmos.DisconnectInternals(breathTool);
+        if (TryComp(toolEntity, out BreathToolComponent? breathTool))
+            _atmos.DisconnectInternals(toolEntity, breathTool);
+
+        if (ent.Comp.BreathTools.Count == 0)
             DisconnectTank(ent);
-        }
 
         _alerts.ShowAlert(ent, ent.Comp.InternalsAlert, GetSeverity(ent));
     }
 
     public void ConnectBreathTool(Entity<InternalsComponent> ent, EntityUid toolEntity)
     {
-        if (TryComp(ent.Comp.BreathToolEntity, out BreathToolComponent? tool))
-        {
-            _atmos.DisconnectInternals(tool);
-        }
-
-        ent.Comp.BreathToolEntity = toolEntity;
+        ent.Comp.BreathTools.Add(toolEntity);
         _alerts.ShowAlert(ent, ent.Comp.InternalsAlert, GetSeverity(ent));
```

Comment:
shouldnt this be a tryadd?

Reply:
that doesn't seem to be a function for Hashsets...

Reply:
and toolEntity doesn't really seem to have a chance to be null?

Reply:
okay, got confused, what i actually meant is that Add returns a bool that you should handle

Reply:
I guess I can do that